src/vs/base/parts/storage/common/storage.ts

```diff
@@ -125,11 +130,11 @@ export class Storage extends Disposable implements IStorage {
 		// items that change external require us to update our
 		// caches with the values. we just accept the value and
 		// emit an event if there is a change.
-		e.changed?.forEach((value, key) => this.accept(key, value));
-		e.deleted?.forEach(key => this.accept(key, undefined));
+		e.changed?.forEach((value, key) => this.accept(key, value, true));
+		e.deleted?.forEach(key => this.accept(key, undefined, true));
 	}
 
-	private accept(key: string, value: string | undefined): void {
+	private accept(key: string, value: string | undefined, wasChangedExternally = false): void {
```

Comment:
No need to set `false` as default as the method is only called with a `boolean`?